frontends/java/src/test/java/ossf/fuzz/introspector/soot/CustomSenceTransformerTest.java

```diff
@@ -0,0 +1,81 @@
+package ossf.fuzz.introspector.soot;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import soot.SceneTransformer;
+
+public class CustomSenceTransformerTest {
+  @Test
+  public void testNoException() {
+    assertDoesNotThrow(() -> {});
+  }
+
+  @Test
+  public void testBasic() {
+    CustomSenceTransformer custom = new CustomSenceTransformer("", "", "");
+    assertTrue(custom instanceof SceneTransformer);
+    assertTrue(custom instanceof CustomSenceTransformer);
+    assertEquals(0, custom.getExcludeList().size());
+  }
+
+  @Test
+  public void testExcludePrefix() {
+    CustomSenceTransformer custom = new CustomSenceTransformer("", "", "abc:def:ghi");
+    assertEquals(custom.getExcludeList().size(), 3);
+    Object[] expected = {"abc", "def", "ghi"};
+    assertArrayEquals(expected, custom.getExcludeList().toArray());
+  }
+
+  @Test
+  public void sampleTestCases() throws IOException, InterruptedException {
```

Comment:
I'm not sure about this test. The problem is that it may become a somewhat blocker of progress. Down the line, if we start to extend how the output looks, then this will fail even though semantics are intact. I feel like the tests should be rooted in semantic rather than syntax -- and this one compares two outputs in a string comparison fashion.

I think the easiest approach is to make the test a bit more narrow and focus on making the tests check that certain properties hold.

WDYT?

Reply:
I agree. And new unitesting has been pushed in PR #663